classes/class-ppc-pagesetups.php

```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
+				$post_column_col = '_posts_custom_column';
+				$ppc_column_data = $manage_col_data . $posttype_col . $post_column_col;
+				add_action( $ppc_column_data, array( $this, 'ppc_column_data_func' ), 10, 2 );
+			}
+		}
+		/**
+		 * Rendered data.
+		 *
+		 * Rendered data into post column.
+		 *
+		 * @param string $columns for column.
+		 * @param string $post_id for post id.
+		 * @since 1.1.0
+		 */
+		public function ppc_column_data_func( $columns, $post_id ) {
+			wp_enqueue_style( 'ppc_backend_css' );
+			$ppc_total_checklist_data = PPC_Loader::get_instance()->get_list();
+			$ppc_checked_data         = get_post_meta( get_the_ID(), '_ppc_meta_key', true );
+			$current_screen_post_type = get_current_screen();
+			$ppc_post_val_array       = isset( $ppc_total_checklist_data[ $current_screen_post_type->post_type ] ) ? $ppc_total_checklist_data[ $current_screen_post_type->post_type ] : '';
+			switch ( $columns ) {
+				case 'ppc_checklist':
+					if ( empty( $ppc_post_val_array ) ) {
+						echo esc_html( sprintf( __( 'No checklist found.', 'pre-publish-checklist' ) ) );
+					} elseif ( isset( $ppc_post_val_array ) && empty( $ppc_checked_data ) ) {
+						echo esc_html( sprintf( __( 'Checklist is empty', 'pre-publish-checklist' ) ) );
+					} elseif ( isset( $ppc_post_val_array ) && ! empty( $ppc_post_val_array ) && isset( $ppc_checked_data ) && ! empty( $ppc_checked_data ) ) {
+						$ppc_result = array_intersect_key( $ppc_post_val_array, $ppc_checked_data );
+						if ( isset( $ppc_result ) && ! empty( $ppc_result ) && count( $ppc_post_val_array ) > count( $ppc_result ) ) {
+							/* translators: %d: number term */
+							echo esc_html( sprintf( __( '%1$d items completed out of %2$d', 'pre-publish-checklist' ), count( $ppc_result ), count( $ppc_post_val_array ) ) );
+							echo '<br>';
+							?>
+						<progress value="<?php echo count( $ppc_result ); ?>" max="<?php echo count( $ppc_post_val_array ); ?>"></progress>
+							<?php
+						} elseif ( count( $ppc_post_val_array ) === count( $ppc_result ) ) {
+							echo esc_html( sprintf( __( 'Checklist is complete', 'pre-publish-checklist' ) ) );
+							echo '<br>';
+							?>
+						<progress value="<?php echo count( $ppc_result ); ?>" max="<?php echo count( $ppc_post_val_array ); ?>"></progress>
+							<?php
+						}
+					}
+					break;
+			}
+		}
+		/**
+		 * Function to add post column.
+		 *
+		 * @param string $columns for column.
+		 *
+		 * @since 1.1.0
+		 */
+		public function ppc_add_column_name_func( $columns ) {
+			$columns = array(
+				'cb'            => '<input type="checkbox"/)',
+				'title'         => 'Title',
+				'author'        => 'Author',
+				'categories'    => 'Categories',
+				'tags'          => 'Tags',
+				'comments'      => '<span class="vers comment-grey-bubble" title="Comments"><span class="screen-reader-text">Comments</span></span>',
+				'date'          => 'Date',
+				'ppc_checklist' => 'Checklist Status',
```

Comment:
The strings here should be translatable.

Reply:
Also, We should just append out the key to array `$columns`. Many post types do not have `author`, `categories` etc. These will get added even if the post type does not support `author`, `categories` etc.

For eg check post type pages.

Reply:
Translation done

Reply:
Extra columns removed. Replace with 
$columns['ppc_checklist'] = __( 'Checklist Status', 'pre-publish-checklist' );